hydra-main/src/main/java/com/addthis/hydra/job/spawn/Spawn.java

```diff
@@ -1260,6 +1265,8 @@ public RebalanceOutcome rebalanceHost(String hostUUID) {
      * @return True if the task is successfully removed
      */
     public boolean deleteTask(String jobUUID, String hostUuid, Integer node, boolean isReplica) {
+        // Maybe - but probably can be better handled by a lock on the individual job/task?
+        // Why do spawnMQ.sendControlMessage and queueJobTaskUpdateEvent need to be guarded?
```

Comment:
At the very least, I think the task.setReplicas call is worthy of consideration.

If a task has replicas A, B, and C, and one thread tries to remove A at the same time as another thread tries to remove B, then it is possible that one removal would be swallowed depending on the timing.